### DISCUSSION TITLE
VRL doesn't have the ability read some filed from a certern file

### DISCUSSION THREAD

**Discussion by lusvan** [vectordotdev/vector#11956] [General]
When i transforms my old log to a standard format log, I need to add many new filed that not exist in the old log file. These new fileds come from other config files. Is the any way to read them?

**Comment by jszwedko** [comment]
Hi @lusvan !

There is an enrichment table feature that might be able to help here. This feature allows reading a CSV file and performing lookups to map fields. See an example here: https://github.com/vectordotdev/vector-demos/tree/main/csv-enrichment

Let me know what you think!

**Comment by lusvan** [comment]
Hi, Thanks for replying
I solve the problem :


1. Parsing log by&nbsp; VRL.


2. Delivering the event to lua transform process.&nbsp;


3. Just perform a pipe system in Lua hookinit to get what I need.
&nbsp;





---Original---
From: "Jesse ***@***.***&gt;
Date: Wed, Mar 23, 2022 23:59 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [vectordotdev/vector] VRL doesn't have the ability read somefiled from a certern file (Discussion #11956)




 
Hi @lusvan !
 
There is an enrichment table feature that might be able to help here. This feature allows reading a CSV file and performing lookups to map fields. See an example here: https://github.com/vectordotdev/vector-demos/tree/main/csv-enrichment
 
Let me know what you think!

**Reply by jszwedko** [reply]
Ah, gotcha, yeah the `lua` transform is another good option.
